File: go/longest-word/longest-word.go
package main

import (
	"fmt"
	"regexp"
)

func longestWord(text string) string {
	pattern := regexp.MustCompile(`[^a-zA-Z]+`)

	words := pattern.Split(text, -1)


	longest := ""
	for _, word := range words {
		if len(word) > len(longest) {
			longest = word
		}
	}

	return longest
}

func main() {
	text := "Ready, steady, go!"
	fmt.Printf("The longest word in the text '%s' is: %s", text, longestWord(text))
}

File: js/longest-word.js
function longestWord(text){
    let pattern = /[^a-z^A-Z]/
    
    text = text.split(pattern)
    
    let longest = text.reduce((a, b) => {
        return a.length > b.length ? a : b
    })
    
    return longest
}

let text = "Ready, steady, go!"

console.log(`The longest word in the text '${text}' is:`, longestWord(text))